core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -480,11 +480,29 @@ public void safeLookup(Object keyBase, long keyOffset, int keyLength, Location l
   }
 
   /**
+   * Append a pair of key-value at the end, there could other pairs with the same keys.
+   */
+  public boolean append(Object keyBase, long keyOffset, int keyLength,
+                        Object valueBase, long valueOffset, int valueLength) {
+    assert(longArray != null);
+    int hash = Murmur3_x86_32.hashUnsafeWords(keyBase, keyOffset, keyLength, 42);
+    int pos = hash & mask;
+    int step = 1;
+    while (longArray.get(pos * 2) != 0) {
```

Comment:
It's the same as we insert new key into the map, the map can be atmost 70% full.